redbox-core/redbox/transform.py

```diff
@@ -50,8 +50,8 @@ def combine_values(field_name: str):
 
 def structure_documents(docs: list[Document]) -> DocumentState:
     return {
-        g_id: {d.metadata["uuid"]: d for d in [d for d in docs if d.metadata["parent_file_uuid"] == g_id]}
-        for g_id in [d.metadata["parent_file_uuid"] for d in docs]
+        g_id: {d.metadata["uuid"]: d for d in [d for d in docs if d.metadata["file_name"] == g_id]}
+        for g_id in [d.metadata["file_name"] for d in docs]
```

Comment:
c175743f5dcedd6dd813f30fe971864ff3b42af5

Reply:
hmmmm actually no, this actually doesnt work. ill rethink

Reply:
this is more subtle than it first appears, `build_merge_pattern` relies upon merging one document into another rather than merging both into a new one, and the tests explicitly check for this behaviour, i.e. https://github.com/i-dot-ai/redbox/blob/68be33cda414e204e41e374dfa04043134191410/redbox-core/tests/graph/test_patterns.py#L184. I would be in favour of changing this to make it symmetrical but in its own PR?